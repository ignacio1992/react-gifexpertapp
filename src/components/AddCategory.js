import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    const [InputValue, setInputValue] = useState('');

    const handleInputChange = (e) =>{
            setInputValue( e.target.value );
    }
    //permite agregar nuevos elementos al arreglo
    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('Submit hecho');
        
        if ( InputValue.trim().length >2 ) {
            setCategories(anime =>[InputValue,...anime] );
            setInputValue('');   
        }
    }
    
    return (
        
        <form onSubmit ={handleSubmit}>
            <input 
             type ='text'
             value={InputValue}
             onChange={handleInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
