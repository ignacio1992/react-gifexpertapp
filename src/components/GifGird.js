import React,{ } from 'react'
import { useFerchGifs } from '../hooks/useFerchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGird = ({category}) => {

    const {data:images,loading} = useFerchGifs(category);

    return (
        <>
        <h3 className='animate__animated animate__fadeIn' > { category}</h3>

            { loading && <p className= 'animate__animated animate__flash'>loading...</p>}

        {<div className ='Card-grid'>
            
                {
                    images.map((img ) =>(
                        <GifGridItem
                         key= {img.id}
                         { ...img }
                         />
                    ))
                }
            </div>}
        </>
    )
}
