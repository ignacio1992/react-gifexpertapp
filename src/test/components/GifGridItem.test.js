import React from 'react'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem'
import PropTypes from 'prop-types';


describe('pruebas en <GifGrindItem', () => {

        const title = 'un titulo';
        const url ='https://localhost/algo.jpg'
        const wrapper = shallow (  <GifGridItem title ={ title}  url ={ url}  />);    
       
        test('debe mostrar el componente correctamente ', () => {
            //no es necesaria debido a que ya existe el mismo elemento fuera..
            // const wrapper = shallow (<GifGridItem title ={ title} url ={ url} />);
            expect (wrapper).toMatchSnapshot();
    
        })

        test('debe de tener un pÃ¡rrafo con el title', () => {

            const p = wrapper.find('p');
            expect(p.text().trim()).toBe( title);
        })
        
        test('debe de tener la imagen igual al url y alt de los props ', () => {
            
                const img =wrapper.find('img');
                //console.log(img.prop('src')); 
        //podemos usar props para buscar varios elementos del objeto a la vez
        //ej console.log(img.props('src','titile'))
                expect(img.prop('src')).toBe(url);
                expect(img.prop('alt')).toBe(title);
                

        })
        

        test('debe de tener animate__bounceIn ', () => {
            
            const div = wrapper.find('div');
            const className = div.prop('className');

            expect( className.includes('animate__bounceIn') ).toBe(true);

        })
         
    
})

