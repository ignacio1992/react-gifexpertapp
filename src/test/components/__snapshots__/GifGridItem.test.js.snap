// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrindItem debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    un titulo
     
  </p>
</div>
`;
